### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by ronnahom96**
Adds S3 as a container for integration tests

| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                       |
| New feature     | ✔                                                                   |
| Breaking change | ✔                                                               |
| Deprecations    | ✖                                                                  |
| Documentation   | ✖                                                                |
| Tests added     | ✔                                                                   |
| Chore            | ✔                                                                      |

### CODE REVIEW

File: docker-compose-db.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: '2'
```
Comment (ronnahom96) [inline docker-compose-db.yml]: remove it

File: tests/integration/ingestion/controllers/ingestionController.spec.ts [hunk 2]
```diff
@@ -10,8 +10,10 @@ import { getProvider } from '../../../../src/providers/getProvider';
 import { createPayload } from '../../../helpers/mockCreator';
 import { IngestionRequestSender } from '../helpers/requestSender';
 
-describe('IngestionController', function () {
+describe('IngestionController on S3', function () {
   let requestSender: IngestionRequestSender;
+  // let s3Helper: S3Helper;
```
Comment (ronnahom96) [inline tests/integration/ingestion/controllers/ingestionController.spec.ts]: remove it

File: tests/integration/ingestion/controllers/ingestionController.spec.ts [hunk 3]
```diff
@@ -31,6 +33,65 @@ describe('IngestionController', function () {
         },
       ],
     });
+    // s3Helper = container.resolve(S3Helper);
```
Comment (ronnahom96) [inline tests/integration/ingestion/controllers/ingestionController.spec.ts]: remove it

File: tests/integration/providers/s3Provider.spec.ts [hunk 4]
```diff
@@ -1,29 +1,35 @@
...
   const queueFilePath = `${os.tmpdir()}/${config.get<string>('ingestion.queueFilePath')}`;
   const s3Config = config.get<IS3Config>('S3');
 
-  beforeAll(() => {
+  beforeAll(async () => {
     getApp({
       override: [
         { token: SERVICES.LOGGER, provider: { useValue: jsLogger({ enabled: false }) } },
         { token: SERVICES.PROVIDER_CONFIG, provider: { useValue: s3Config } },
+        // { token: SERVICES.QUEUE_FILE_HANDLER, provider: { useValue: queueFileHandlerMock } },
```
Comment (ronnahom96) [inline tests/integration/providers/s3Provider.spec.ts]: remove comment

File: tests/integration/providers/s3Provider.spec.ts [hunk 5]
```diff
@@ -34,23 +40,35 @@ describe('S3Provider tests', () => {
...
+    s3Helper.killS3();
+  });
+
+  describe('streamModelPathsToQueueFile', () => {
     it('returns all the files from S3', async () => {
-      const model = 'model1';
-      const expected = `${model}/file1\n${model}/file2\n`;
-      s3Mock.on(ListObjectsCommand).resolvesOnce(s3Output);
+      const model = randWord();
+      const lengthOfFiles = randNumber({ min: 1, max: 5 });
```
Comment (ronnahom96) [inline tests/integration/providers/s3Provider.spec.ts]: fileLength would be more appropriate name.

**Review by netanelC — APPROVED** [review]

**Pull request merged**